### PR TITLE
Fix crash on exit. 

### PR DESCRIPTION
**Pull request by jimfcarroll**
This crash seems to be due to the logger being set on CThreads (and Exception) using XbmcContext rather than the means prescribed by GlobalHandling. 

## Description

XbmcContext was originally supposed to be the "down payment" on a dependency injection approach to modularizing the code but it never went anywhere so it's now removed all together. All of the code that was dependent on this functionality has been reverted to the standard means. 

## Motivation and Context

Crash on exit.

## How Has This Been Tested?

Ran it several time on Linux so far to make sure it doesn't crash. Since the crash was rare this isn't a foolproof means of testing. @FernetMenta was able to reproduce it relatively easily.

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by FernetMenta** [comment]
thanks!
+25 −164
:)

File: xbmc/commons/Exception.cpp [hunk 1]
```diff
@@ -19,11 +19,15 @@
...
 
 namespace XbmcCommons
 {
-  ILogger* Exception::logger = NULL;
-
   Exception::~Exception() = default;
+
+  void Exception::LogThrowMessage(const char* prefix) const
+  {
+  	CLog::Log(LOGERROR,"EXCEPTION Thrown (%s) : %s", classname.c_str(), message.c_str());
```
Comment (Rechi) [inline xbmc/commons/Exception.cpp]: indentation

File: xbmc/platform/posix/main.cpp [hunk 2]
```diff
@@ -88,7 +87,6 @@ void XBMC_POSIX_HandleSignal(int sig)
 int main(int argc, char* argv[])
 {
   // set up some xbmc specific relationships
```
Comment (Rechi) [inline xbmc/platform/posix/main.cpp]: comment still valid?

File: xbmc/platform/win32/WinMain.cpp [hunk 3]
```diff
@@ -127,7 +126,6 @@ INT WINAPI WinMain(HINSTANCE hInst, HINSTANCE, LPSTR commandLine, INT)
   int status;
   {
     // set up some xbmc specific relationships
```
Comment (Rechi) [inline xbmc/platform/win32/WinMain.cpp]: comment still valid?

File: xbmc/threads/Thread.cpp [hunk 4]
```diff
@@ -226,7 +223,7 @@ void CThread::Action()
   }
   catch (...)
   {
-    LOG(LOGERROR, "%s - thread %s, Unhandled exception caught in thread process, aborting. auto delete: %d", __FUNCTION__, m_ThreadName.c_str(), IsAutoDelete());
+  CLog::Log(LOGERROR, "%s - thread %s, Unhandled exception caught in thread process, aborting. auto delete: %d", __FUNCTION__, m_ThreadName.c_str(), IsAutoDelete());
```
Comment (Rechi) [inline xbmc/threads/Thread.cpp]: indentation

File: xbmc/commons/Exception.cpp (hunk 1 above)
Comment (jimfcarroll) [reply xbmc/commons/Exception.cpp]: hummm. That's odd. I thought I had that right. Will fix.

**Comment by jimfcarroll** [comment]
Okay, Dealt with the issues that @Rechi pointed out.

**Comment by jimfcarroll** [comment]
Commits now squashed

**Pull request merged**